pkg/retry/azure_error.go

```diff
@@ -262,22 +262,35 @@ func getRetryAfter(resp *http.Response) time.Duration {
 	return dur
 }
 
-// GetErrorWithRetriableHTTPStatusCodes gets an error with RetriableHTTPStatusCodes.
-// It is used to retry on some HTTPStatusCodes.
-func GetErrorWithRetriableHTTPStatusCodes(resp *http.Response, err error, retriableHTTPStatusCodes []int) *Error {
-	rerr := GetError(resp, err)
+// IsInHTTPStatusCodeSet return true when status code falls in the status code list
+// It is used with doBackoffRetry to retry on some HTTPStatusCodes.
+func IsInHTTPStatusCodeSet(rerr *Error, HTTPStatusCodes []int) bool {
 	if rerr == nil {
-		return nil
+		return false
 	}
-
-	for _, code := range retriableHTTPStatusCodes {
+	for _, code := range HTTPStatusCodes {
```

Comment:
nit: function variable needs to start with a lower case

Reply:
updated